### PR TITLE
[Feature] Add a possibility to hide a column in the attribute table

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

**Comment by 3nids** [comment]
it seems the filtering is achieved via the attribute table.
Shouldn't there be an original definition in the vector layer properties (which might be overwritten on-the-fly in the attribute table)?

**Comment by alexbruy** [comment]
Maybe I misunderstand the purpose of the PR, but we already can make a field hidden by changing its type in the layer properties dialog.

**Comment by fjacon** [comment]
With this PR, end user get a more friendly UI when they only need to edit some attributes.

**Comment by sbrunner** [comment]
@3nids I'm afraid to have side effects in place of just using a QT method ...

**Comment by sbrunner** [comment]
Up, for a review...

**Comment by sbrunner** [comment]
Test fixed :-)

File: src/app/qgsattributetabledialog.cpp [hunk 1]
```diff
@@ -71,6 +72,12 @@ QgsAttributeTableDialog::QgsAttributeTableDialog( QgsVectorLayer *theLayer, QWid
 {
   setupUi( this );
 
+  const QgsFields layerAttributes = mLayer->fields();
```
Comment (m-kuhn) [inline src/app/qgsattributetabledialog.cpp]: FYI, we have now

    Q_FOREACH( const QgsField& field, mLayer->fields() )

File: src/gui/attributetable/qgsfiltertablefieldsdialog.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+class GUI_EXPORT QgsFilterTableFieldsDialog : public QDialog, private Ui::QgsFilterTableFieldsDialog
+{
+    Q_OBJECT
+
+  public:
+    /**
+     * Constructor
+     * @param vl The concerned vector layer
+     * @param visible the cuurent list of visible fields name
```
Comment (m-kuhn) [inline src/gui/attributetable/qgsfiltertablefieldsdialog.h]: typo

**Comment by m-kuhn** [comment]
Code looks good to me, haven't tested it though. The only thing I'm a bit worried is about cluttering the attribute table with yet one more button.

Maybe we should start hiding away some of the tools in grouped dropdown tool buttons or a menu.

File: src/app/qgsattributetabledialog.cpp (hunk 1 above)
Comment (sbrunner) [inline src/app/qgsattributetabledialog.cpp]: Better effectively, thanks :-)

File: src/gui/attributetable/qgsfiltertablefieldsdialog.h (hunk 2 above)
Comment (sbrunner) [inline src/gui/attributetable/qgsfiltertablefieldsdialog.h]: fixed

**Comment by sbrunner** [comment]
> Maybe we should start hiding away some of the tools in grouped dropdown tool buttons or a menu.

I completely agree, I think that we should defined what we relay want to do..., shouldn't be in an other pull request?

**Comment by m-kuhn** [comment]
Is it on purpose that this is not persisted when the attribute table is closed?

**Comment by m-kuhn** [comment]
@sbrunner ping

I am currently working on a possibility to manually order the table columns. I would build on this dialog if possible, but the changes will need to be persisted in the .qgs project, so it would be good to know if that would fit with your plans.

**Comment by sbrunner** [comment]
OK, thank I think that good if everything will be persistent, then do we merge this pull request, and do you make it persistent, or should I modify this pull request?

**Comment by m-kuhn** [comment]
I think to introduce the ordering we will need to change the code a bit. `setColumnHidden` with the assumption that `fieldIndex == columnIndex` will no longer work. Instead the filtering and reordering will have to take place on the proxy model. So I would say it's better to finish these tasks before merging.

**Comment by sbrunner** [comment]
OK, I plane to do it next week!

**Comment by m-kuhn** [comment]
I'm working on it right now. I'll send you a message when it's ready for review, ok?

**Comment by sbrunner** [comment]
OK, thanks :-)

**Comment by m-kuhn** [comment]
Integrated into #3010 (Not yet ready for review)

**Comment by Gustry** [comment]
Thanks for this feature, but it seems that I have some bugs. Should I open a ticket @sbrunner ?

**Pull request closed without merging**